codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerInstantiator.kt

```diff
@@ -47,12 +48,26 @@ class ServerBuilderKindBehavior(val codegenContext: CodegenContext) : Instantiat
     override fun hasFallibleBuilder(shape: StructureShape): Boolean {
         // Only operation input builders take in unconstrained types.
         val takesInUnconstrainedTypes = shape.isReachableFromOperationInput()
-        return ServerBuilderGenerator.hasFallibleBuilder(
-            shape,
-            codegenContext.model,
-            codegenContext.symbolProvider,
-            takesInUnconstrainedTypes,
-        )
+
+        val publicConstrainedTypes = if (codegenContext is ServerCodegenContext) {
```

Comment:
This is correct, although `codegenContext` should always be of type `ServerCodegenContext` here. I don't understand why we're invoking the server instantiator when `publicConstrainedTypes` is `false` though.